internal/services/containers/kubernetes_cluster_node_pool_resource_test.go

```diff
@@ -2706,3 +2721,38 @@ resource "azurerm_kubernetes_cluster_node_pool" "test" {
 }
  `, data.Locations.Primary, data.RandomInteger, data.RandomString)
 }
+
+func (KubernetesClusterNodePoolResource) gpuInstance(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-aks-%[2]d"
+  location = "%[1]s"
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%[2]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%[2]d"
+  default_node_pool {
+    name       = "default"
+    node_count = 1
+    vm_size    = "Standard_D2s_v3"
+  }
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_kubernetes_cluster_node_pool" "test" {
+  name                  = "internal"
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.test.id
+  vm_size               = "Standard_ND96asr_v4"
```

Comment:
Can we please use a smaller, less expensive vm. We don't need to spin up a vm with 96 vCPUs and 900 GB memory to test this?

Reply:
Got it! I've changed the vmsize.